virttest/ip_sniffing.py

```diff
@@ -0,0 +1,272 @@
+"""
+IP sniffing facilities
+"""
+
+import threading
+import logging
+import re
+
+import aexpect
+from avocado.utils import path as utils_path
+
+from remote import handle_prompts
+from utils_misc import log_line
+from utils_misc import format_str_for_message
+
+
+class AddrCache(object):
```

Comment:
This could have been significantly simpler if you'd used `dict`, but it looks like complete implementation of mapping so so far so good.